app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardItemViewModel.kt

```diff
@@ -0,0 +1,217 @@
+package org.wikipedia.feed.suggestededits
+
+import android.os.Bundle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withTimeoutOrNull
+import org.wikipedia.Constants
+import org.wikipedia.WikipediaApp
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.dataclient.mwapi.MwQueryPage
+import org.wikipedia.descriptions.DescriptionEditActivity
+import org.wikipedia.page.Namespace
+import org.wikipedia.page.PageTitle
+import org.wikipedia.suggestededits.PageSummaryForEdit
+import org.wikipedia.suggestededits.provider.EditingSuggestionsProvider
+import org.wikipedia.util.Resource
+import org.wikipedia.util.StringUtil
+import java.net.SocketTimeoutException
+
+class SuggestedEditsCardItemViewModel(bundle: Bundle) : ViewModel() {
+
+    val age = bundle.getInt(SuggestedEditsCardItemFragment.EXTRA_AGE)
+    var cardActionType = bundle.getSerializable(SuggestedEditsCardItemFragment.EXTRA_ACTION_TYPE) as DescriptionEditActivity.Action
+    var sourceSummaryForEdit: PageSummaryForEdit? = null
+    var targetSummaryForEdit: PageSummaryForEdit? = null
+    var imageTagPage: MwQueryPage? = null
+
+    private val appLanguages = WikipediaApp.instance.languageState.appLanguageCodes
+
+    private val _uiState = MutableStateFlow(Resource<Boolean>())
+    val uiState = _uiState.asStateFlow()
+
+    init {
+        fetchCardData()
+    }
+    fun fetchCardData() {
+        _uiState.value = Resource.Loading()
+        viewModelScope.launch(CoroutineExceptionHandler { _, throwable ->
+            // Give retry option to user in case of network error
+            _uiState.value = Resource.Error(SocketTimeoutException(throwable.message))
```

Comment:
Is there a reason for wrapping the `throwable` in a SocketTimeoutException? (Can we just pass the `throwable` itself?)

Reply:
Done.